controller/Filters/ListFilter.js

```diff
@@ -14,60 +20,83 @@ const listFilter = async (req, res, next) => {
                 }
             })
+            // const moviepersonrole=await models.MoviePersonRole.findAll({
+            //     where:{
+            //         moviePersonId:movieperson.id
+            //     }
+            // })
             obj = [...JSON.parse(JSON.stringify(movieperson, null, 4))];
             person = []
-            //console.log(obj)
+            console.log(obj)
             for (i = 0; i < obj.length; i++) {
-                if (req.query.age) {
-                    const actordetails = await models.Person.findOne({
-                        where: {
-                            id: obj[i].personId,
-                            age: req.query.age
-                        }
-                    })
-                    if (actordetails)
-                        person.push(actordetails.name)
-                }
-                else {
-                    const actordetails1 = await models.Person.findOne({
-                        where: {
-                            id: obj[i].personId,
-                        }
-                    })
-                    person.push(actordetails1.name)
+                const moviepersonrole = await models.MoviePersonRole.findOne({
+                    where: {
+                        moviePersonId: obj[i].id,
+
+                    }
+                })
+                if (moviepersonrole.roleId == 1 || moviepersonrole.roleId == 2) {
+                    if (req.query.age) {
+
+                        const actordetails = await models.Person.findOne({
+                            where: {
+                                id: obj[i].personId,
+                                age: req.query.age,
+
+                            }
+                        })
+                        if (actordetails)
+                            person.push(actordetails.name)
+                    }
+                    else {
+
+                        const actordetails1 = await models.Person.findOne({
+                            where: {
+                                id: obj[i].personId,
+                            }
+                        })
+                        person.push(actordetails1.name)
+                    }
                 }
+                c = 1
+
             }
-            c = 1
 
         }
-        if (req.query.age) {
+        else if (req.query.age) {
+            person = []
             persons = await models.Person.findAll({
                 where: {
                     age: req.query.age
                 }
             })
+            obj1 = [...JSON.parse(JSON.stringify(persons, null, 4))];
+            console.log(obj1)
+            obj1.map(item => {
+                person.push(item.name)
+            })
             a = 1
 
             //res.status(200).json({persons})
 
         }
 
-
         if (c == 1 && a == 1) {
             res.status(200).json({ person })
         }
         else if (c == 1 && a == 0) {
             res.status(200).json({ person })
```

Comment:
Write one common function to handle response, and call that method to return response by sending statusCode and result
For success response:{ success: true, data: result }
For failure:{ sucess: false, error: actual error}

Reply:
yes,return in one function and called those in required functions